core/arch/arm/mm/core_mmu.c

```diff
@@ -444,6 +436,28 @@ static bool __maybe_unused is_map_tee_ram(struct tee_mmap_region *mm)
 	return (mm->type == MEM_AREA_TEE_RAM);
 }
 
+static bool is_map_flat_mapped(struct tee_mmap_region *mm)
+{
+	return is_map_tee_ram(mm);
+}
+
+static bool __maybe_unused is_map_secure(struct tee_mmap_region *mm)
+{
+	return !!(core_mmu_type_to_attr(mm->type) & TEE_MATTR_SECURE);
+}
+
+static void swap_mmap(struct tee_mmap_region *mm1, struct tee_mmap_region *mm2)
+{
+	struct tee_mmap_region swap;
+
+	if (mm1 == mm2)
+		return;
+
+	memcpy(&swap, mm1, sizeof(swap));
```

Comment:
This could be written as (which is safer and also easier to read in my opinion):
```
swap = *mm1;
*mm1 = *mm2;
*mm2 = swap;
```